docs/development_guide.md

```diff
@@ -203,6 +203,19 @@ you will find the per-function analysis useful. This will show you code paths th
 
 ![per function](./images/unit-test-coverage-3.png)
 
+##### Mocking
+
+OSM uses the [GoMock](https://github.com/golang/mock) mocking framework to mock interfaces in unit tests.
+GoMock's `mockgen` tool is used to autogenerate mocks from interfaces.
+
+As an example, to create a mock client for the `Configurator` interface defined in the [configurator](/pkg/configurator) package:
+```bash
+mockgen -destination=pkg/configurator/mock_client.go -package=configurator github.com/openservicemesh/osm/pkg/configurator Configurator
+```
```

Comment:
Could you either add to this PR or follow up with a change to include this package in tools.go at the root of the repo and update the instructions to use `go run ...` to invoke the generator? That way we can track the version we're using and there's no extra step required to get the `mockgen` binary.

Reply:
Will do in a follow-up.